### ISSUE TITLE
[NVIDIA] Steam BPM too slow when started from the default steam.desktop

### ISSUE THREAD

**Issue by brunocalmon** [ublue-os/bazzite#1160]
### Describe the bug

The Steam BPM interface is too laggy, however it gets smooth if I simply create another .desktop for it with only 'exec= steam' param.

laggy:

https://github.com/ublue-os/bazzite/assets/8751648/98a33219-baa0-43a4-9d02-0f1e54782fc6

fixed:

https://github.com/ublue-os/bazzite/assets/8751648/a34e0bfe-c023-441b-b523-c0871c18f5c4

### What did you expect to happen?

Steam running smooth out of the box.

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-asus-nvidia:stable
                   Digest: sha256:82593a8a1bf98d58fe46d30bd0760d3d50348cc84d7a28ea42ffda5dd3be4634
                  Version: 40.20240521.0 (2024-05-22T06:00:40Z)
                     Diff: 1 added
          LayeredPackages: zsh

● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-asus-nvidia:stable
                   Digest: sha256:82593a8a1bf98d58fe46d30bd0760d3d50348cc84d7a28ea42ffda5dd3be4634
                  Version: 40.20240521.0 (2024-05-22T06:00:40Z)
               LiveCommit: 1a3804d6a5babc32e56372daba1df43e40b48b7fa97ee5ba7478ebad145a7296
                 LiveDiff: 1 added
                 Unlocked: transient

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-asus-nvidia:stable
                   Digest: sha256:033593ccef49c15f10af9c489162eb5ca41d393d2e32ec6a20a7b21c5353183c
                  Version: 40.20240521.0 (2024-05-21T17:04:08Z)
```

### Hardware

# System Details Report
---

## Report details
- **Date generated:**                              2024-05-22 15:01:45

## Hardware Information:
- **Hardware Model:**                              ASUSTeK COMPUTER INC. ROG Zephyrus G15 GA503RS_GA503RS
- **Memory:**                                      16.0 GiB
- **Processor:**                                   AMD Ryzen™ 9 6900HS with Radeon™ Graphics × 16
- **Graphics:**                                    AMD Radeon™ Graphics
- **Graphics 1:**                                  NVIDIA GeForce RTX™ 3080 Laptop GPU
- **Disk Capacity:**                               2.0 TB

## Software Information:
- **Firmware Version:**                            GA503RS.317
- **OS Name:**                                     Fedora Linux 40.20240521.0 (Bazzite GNOME)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               46
- **Windowing System:**                            Wayland
- **Kernel Version:**                              Linux 6.8.9-301.fsync.fc40.x86_64

### Extra information or context

This problem happens in all version KDE or GNOME using wayland or x11, it doesn't matter the combination.

But I'm currently using GNOME+Wayland because for some reason X11 is not working for external display, the whole system gets unusable due to a very slow performance. It's another issue but I believe it worths to mentioned that only happens if I set the external display to primary and disable to built-in, if I keep the built-in enabled the system runs smooth as well.

**Comment by HikariKnight** [comment]
what is the output of `switcherooctl list | grep -o 'Device:' | wc -l` on your laptop
ideally it should be `1`

**Comment by Zeglius** [comment]
My guess is that lag had something to do with the '-cef-disable-gpu' flag always being added. Check #1170. That should fix it whenever gets included in the images.

Checked in my hybrid Intel/Nvidia laptop executing `steam -tenfoot` and `steam -tenfoot -cef-disable-gpu`. The later feels incredibly laggy, the former is smoother.

TL;DR: Whenever #1170 lands in stable, my guess is this should be fixed.

**Comment by erdii** [comment]
Both BPM (Big Picture Mode) and the regular steam UI lag a lot on my bazzite installation too when started via the default launcher even though `-cef-disable-gpu` does not seem to be in use. (`ps aux | grep -i cef-disable-gpu` does not print any process besides the command itself).

When I run `steam` directly from the terminal, everything seems to be working fine (no stutter in game trailers in the shop and no lag in BPM).

Output of `rpm-ostree status`:

```
erdii@fedora:~$ rpm-ostree status
State: idle
Deployments:
  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-nvidia:stable
                   Digest: sha256:88481ac59982d1e568953241b155c404b228029ac68ce311476c01aee5b31e8d
                  Version: 40.20240603.0 (2024-06-03T23:25:55Z)
                     Diff: 3 upgraded
          LayeredPackages: mullvad-vpn
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf" ' 

● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-nvidia:stable
                   Digest: sha256:21d358ad2948035084dfb14b9da6f2595638d7185c68cc69938c44374d4e0923
                  Version: 40.20240603.0 (2024-06-03T18:09:18Z)
          LayeredPackages: mullvad-vpn
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf" ' 

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-gnome-nvidia:stable
                   Digest: sha256:ce071dd4d513c1dd191d6ea3f9652c7d04126ba4798f886f73bda6ab803e98ca
                  Version: 40.20240531.0 (2024-05-31T17:04:28Z)
          LayeredPackages: mullvad-vpn
                Initramfs: '"-I /etc/crypttab /usr/lib/modprobe.d/nvidia.conf" ' 
```

# System Details Report
---

## Report details
- **Date generated:**                              2024-06-04 19:22:32

## Hardware Information:
- **Hardware Model:**                              Micro-Star International Co., Ltd. MS-7B93
- **Memory:**                                      32,0 GiB
- **Processor:**                                   AMD Ryzen™ 7 3700X × 16
- **Graphics:**                                    NVIDIA GeForce RTX™ 3070
- **Disk Capacity:**                               2,5 TB

## Software Information:
- **Firmware Version:**                            1.80
- **OS Name:**                                     Fedora Linux 40.20240603.0 (Bazzite GNOME)
- **OS Build:**                                    (null)
- **OS Type:**                                     64-bit
- **GNOME Version:**                               46
- **Windowing System:**                            X11
- **Kernel Version:**                              Linux 6.8.11-301.fsync.fc40.x86_64

**Comment by KyleGospo** [comment]
Known nvidia issue with BPM, I would take this to Valve's github as others have. We have no power to fix this.

**Comment by erdii** [comment]
Huh. That's unfortunate. Thanks for the feedback @KyleGospo!

I went over to [ValveSoftware/steam-for-linux/issues](https://github.com/ValveSoftware/steam-for-linux/issues) and tried to search a similar reproducer (including the lags going away when executing plain `steam`) and there are really a TON of issues, some of them being a couple years old, with lags in the regular UI and BPM on a whole range of distro/hardware combinations.

I tried to nail one down via these search terms but I couldn't find one:
- [open lag terminal](https://github.com/ValveSoftware/steam-for-linux/issues?q=is%3Aissue+is%3Aopen+lag+terminal)
- [open lag shell](https://github.com/ValveSoftware/steam-for-linux/issues?q=is%3Aissue+is%3Aopen+lag+shell)
- [open lag](https://github.com/ValveSoftware/steam-for-linux/issues?q=is%3Aissue+is%3Aopen+lag)

If you could point me to a specific issue, I'd be happy to add my observations there but given the amount of rotting issues I wouldn't expect anything happening from valve's side soon.

**What fixes it for me on bazzite right now, as of today**, is using a modified `.desktop` launcher file, stored in `~/.local/share/applications/steam-fixed.desktop`, that just runs `steam` instead of `/usr/bin/bazzite-steam %U`:

`diff /usr/share/applications/steam.desktop ~/.local/share/applications/steam-fixed.desktop`
```diff 
2,3c2,3
< Name=Steam
< Comment=Application for managing and playing games on Steam
---
> Name=Steam (fixed)
> Comment=Application for managing and playing games on Steam (fixed launcher - no lagging)
30c30
< Exec=/usr/bin/bazzite-steam %U
---
> Exec=steam
```

**Comment by cosmin-bianu** [comment]
I've had this issue too. I've just managed to fix it on an Intel/Nvidia laptop by enabling GPU acceleration in Steam. For some reason, Bazzite uses the integrated Intel GPU by default. 

1. Within Steam Gaming Mode, press the **Menu** button
2. Go down to **Power** and select **Switch to Desktop**
3. Open the Steam desktop client (by default it starts automatically after a few moments)
4. In the top left, go to **Steam** > **Settings**
5. Under **Interface**, scroll down and enable the two options related to hardware acceleration. As of writing this, the settings are called:
- **Enable GPU accelerated rendering in web views (requires restart)**
- **Enable hardware video decoding, if supported (requires restart)**
6. Restart the Steam client and ensure the two options are still ticked. If so, use the **Return to Gaming Mode** shortcut on your desktop and test whether this works for you.

Hope this helps.

**Comment by mdunlap** [comment]
Revisiting this with new evidence — there is actually a launch flag in the Steam binary that fixes the BPM sluggishness on NVIDIA, and I think it warrants reconsideration.                 
                                                                  
The flag: -cef-force-gpu                                                                       
                                                                  
It's compiled into /usr/bin/steam (visible in the strings table alongside -cef-disable-gpu,  -cef-disable-sandbox, -cef-force accessibility, etc.) and tells Steam to flip the CEF launch arguments from --disable-gpu --disable-gpu-compositing to: ignore-gpu-blocklist. This enables GPU acceleration in steamwebhelper without going through the Steam Settings → Interface toggle.
